src/borg/archiver.py

```diff
@@ -1079,14 +1079,14 @@ def _delete_archives(self, args, repository):
             deleted = False
             for i, archive_name in enumerate(archive_names, 1):
                 try:
-                    del manifest.archives[archive_name]
+                    current_archive = format_archive(manifest.archives.pop(archive_name))
```

Comment:
better do the format_archive() call later (see below).

when doing it here, the exception handler here covers more code than wanted and it would be bad if a KeyError would be raised inside format_archive code and you catch it here also.